FaunaDB.Client/Types/RefV.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using FaunaDB.Query;
+using Newtonsoft.Json;
+
+namespace FaunaDB.Types
+{
+    /// <summary>
+    /// A FaunaDB ref type.
+    /// <para>
+    /// See <see href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</see>.
+    /// </para>
+    /// </summary>
+    public class RefV : Value
+    {
+        public string Id { get; }
+        public RefV Class { get; }
+        public RefV Database { get; }
```

Comment:
this may be my ignorance of the recursive ref spec, but is `Class` and `Database` here always a `ClassV` and a `DatabaseV`? Or will they be arbitrary RefVs?